### PR TITLE
kenlm: add penalty for impossible Māori words

### PR DESCRIPTION
**Pull request by douglasbagnall**
I think this makes words containing consonant clusters considered 10,000 times less likely by the language model (compared to other out-of-vocab words).

### CODE REVIEW

**Comment by edwardabraham** [comment]
Thanks, that's great. Will park this until we are ready to compile it and test it out :)

File: native_client/deepspeech.cc [hunk 1]
```diff
@@ -150,10 +150,12 @@ DEEPSPEECH_EXPORT
 void
 Model::enableDecoderWithLM(const char* aAlphabetConfigPath, const char* aLMPath,
                            const char* aTriePath, float aLMWeight,
-                           float aWordCountWeight, float aValidWordCountWeight)
+                           float aWordCountWeight, float aValidWordCountWeight,
+                           float InalidWordPenalty)
```
Comment (mathematiguy) [inline native_client/deepspeech.cc]: Spelling error here - should be `aInvalidWordPenalty`

File: native_client/deepspeech.h [hunk 2]
```diff
@@ -68,7 +68,8 @@ namespace DeepSpeech
                                const char* aLMPath, const char* aTriePath,
                                float aLMWeight,
                                float aWordCountWeight,
-                               float aValidWordCountWeight);
+                               float aValidWordCountWeight,
+                               float InvalidWordPenalty);
```
Comment (mathematiguy) [inline native_client/deepspeech.h]: Likewise here- I think

**Comment by mathematiguy** [comment]
Finally started compiling these changes, at the moment the compile fails with the following errors:

```
native_client/beam_search.h: In function 'bool IsMaoriWord(std::string)':
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
   };
   ^
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
native_client/beam_search.h:36:3: sorry, unimplemented: non-trivial designated initializers not supported
INFO: Elapsed time: 927.356s, Critical Path: 87.59s
FAILED: Build did NOT complete successfully
makefile:40: recipe for target 'build_native_client' failed
make: *** [build_native_client] Error 1
```

**Comment by mathematiguy** [comment]
I did the obvious and looked up the error on StackOverflow, where I found this: https://stackoverflow.com/questions/31215971/non-trivial-designated-initializers-not-supported

**Comment by douglasbagnall** [comment]
> I did the obvious and looked up the error on StackOverflow, where I found this: https://stackoverflow.com/questions/31215971/non-trivial-designated-initializers-not-supported

The first answer is probably right -- I wrote C when it wanted C++, so 

>  Remove the "." and change "=" to ":" when initializing.

**Comment by douglasbagnall** [comment]
Also, I said "consonant clusters considered 10,000 times less likely", but it looks like I meant 100,000 times, because the penalty is -5 on a log 10 scale.

**Comment by mathematiguy** [comment]
This is acceptable to me lol. I’ve already got a different error now, so at least it’s chugging along

**Comment by utunga** [comment]
Lets just accept we aint gonna do this

**Pull request closed without merging**